Author: csagan5 <32685696+csagan5@users.noreply.github.com>
Date:   Wed Dec 20 22:31:37 2017 +0100

    Revert "Disable runtime flags for /deep/ and ::shadow used in CSS dynamic profile for M63."
    
    This reverts commit 0ebd56da93abf970128804390879624b3dc79ac2.

diff --git a/third_party/WebKit/LayoutTests/fast/dom/shadow/apply-deep-in-document-scope-expected.txt b/third_party/WebKit/LayoutTests/fast/dom/shadow/apply-deep-in-document-scope-expected.txt
--- a/third_party/WebKit/LayoutTests/fast/dom/shadow/apply-deep-in-document-scope-expected.txt
+++ b/third_party/WebKit/LayoutTests/fast/dom/shadow/apply-deep-in-document-scope-expected.txt
@@ -1,4 +1,4 @@
-CONSOLE WARNING: line 1: /deep/ combinator is no longer supported in CSS dynamic profile.It is now effectively no-op, acting as if it were a descendant combinator. /deep/ combinator will be removed, and will be invalid at M65. You should remove it. See https://www.chromestatus.com/features/4964279606312960 for more details.
+CONSOLE WARNING: line 1: /deep/ combinator in CSS is deprecated and will be removed in M63, around December 2017. See https://www.chromestatus.com/features/4964279606312960 for more details.
 /deep/ as a descendant selector in document without shadow trees.
 
 On success, you will see a series of "PASS" messages, followed by "TEST COMPLETE".
diff --git a/third_party/WebKit/LayoutTests/shadow-dom/v0/closed-mode-deep-combinator-and-shadow-pseudo-expected.txt b/third_party/WebKit/LayoutTests/shadow-dom/v0/closed-mode-deep-combinator-and-shadow-pseudo-expected.txt
--- a/third_party/WebKit/LayoutTests/shadow-dom/v0/closed-mode-deep-combinator-and-shadow-pseudo-expected.txt
+++ b/third_party/WebKit/LayoutTests/shadow-dom/v0/closed-mode-deep-combinator-and-shadow-pseudo-expected.txt
@@ -1,4 +1,4 @@
-CONSOLE WARNING: line 311: /deep/ combinator is no longer supported in CSS dynamic profile.It is now effectively no-op, acting as if it were a descendant combinator. /deep/ combinator will be removed, and will be invalid at M65. You should remove it. See https://www.chromestatus.com/features/4964279606312960 for more details.
+CONSOLE WARNING: line 311: /deep/ combinator in CSS is deprecated and will be removed in M63, around December 2017. See https://www.chromestatus.com/features/4964279606312960 for more details.
 (1/6) /deep/ style rule on top-level document.
 PASS backgroundColorOf('host_open_open') is "rgba(0, 0, 0, 0)"
 PASS backgroundColorOf('host_open_open/div1') is "rgba(0, 0, 0, 0)"
diff --git a/third_party/WebKit/Source/core/frame/Deprecation.cpp b/third_party/WebKit/Source/core/frame/Deprecation.cpp
--- a/third_party/WebKit/Source/core/frame/Deprecation.cpp
+++ b/third_party/WebKit/Source/core/frame/Deprecation.cpp
@@ -388,13 +388,13 @@ DeprecationInfo GetDeprecationInfo(WebFeature feature) {
               "details.")};
 
     case WebFeature::kCSSDeepCombinator:
-      return {"CSSDeepCombinator", M65,
-              "/deep/ combinator is no longer supported in CSS dynamic profile."
-              "It is now effectively no-op, acting as if it were a descendant "
-              "combinator. /deep/ combinator will be removed, and will be "
-              "invalid at M65. You should remove it. See "
-              "https://www.chromestatus.com/features/4964279606312960 for more "
-              "details."};
+      return {"CSSDeepCombinator", M63,
+              willBeRemoved("/deep/ combinator in CSS", M63, "4964279606312960")};
+
+    case WebFeature::kCSSSelectorPseudoShadow:
+      return {"CSSSelectorPseudoShadow", M63,
+              willBeRemoved("::shadow pseudo-element in CSS", M63,
+                           "6750456638341120")};
 
     case WebFeature::kVREyeParametersOffset:
       return {"VREyeParametersOffset", Unknown,
diff --git a/third_party/WebKit/Source/platform/runtime_enabled_features.json5 b/third_party/WebKit/Source/platform/runtime_enabled_features.json5
--- a/third_party/WebKit/Source/platform/runtime_enabled_features.json5
+++ b/third_party/WebKit/Source/platform/runtime_enabled_features.json5
@@ -325,10 +325,11 @@
       name: "DecodeToYUV",
       status: "experimental",
     },
-    // Remove this flag once we can remove /deep/ at M63.
+    // Introduced this flag as stable so we can re-enable this feature easily
+    // after we disable this feature.
     {
       name: "DeepCombinatorInCSSDynamicProfile",
-      status: "test",
+      status: "stable",
     },
     {
       name: "DeprecationReporting",
@@ -978,10 +979,11 @@
       name: "ShadowPiercingDescendantCombinator",
       status: "experimental",
     },
-    // Remove this flag once we can remove ::shadow at M63.
+    // Introduced this flag as stable so we can re-enable this feature easily
+    // after we disable this feature.
     {
       name: "ShadowPseudoElementInCSSDynamicProfile",
-      status: "test",
+      status: "stable",
     },
     {
       name: "ShapeDetection",
